frontend/src/slices/trucks/trucks.slice.ts

```diff
@@ -90,7 +90,16 @@ const { reducer, actions, name } = createSlice({
   extraReducers(builder) {
     builder
       .addCase(addTruck.fulfilled, (state, action) => {
-        state.trucks.unshift(action.payload);
+        const newTruck = action.payload;
+
+        const isTruckUnique = !state.trucks.some(
+          (truck) => truck.id === newTruck.id,
+        );
```

Comment:
I don't understand this check here. 
Am I right that on be we can create new duplicated truck, and on fe we are checking it after successfull saving on be? 🤔

Reply:
No, there is no duplication on be. There is duplication only on the UI. I use **unshift** so that I can add a truck to the top of the table on the page where the user is now. However, there may be such a situation that the page has been refreshed and due to sorting this added truck is already on this page, and we are on the UI, and using **unshift** we add it to the top of the table, and it looks like we have two identical trucks on one page, but this is only on the UI. When the page is reloaded, this additional truck will disappear. That's why I added this check to avoid such a case. If the page has already received a truck, we do not add it to the top, and if the truck is to be added to another page on be, and not to the current one, we temporarily move it to the top on fe so that the user can see that he has created a truck.